packages/devtools/lib/src/inspector/inspector.dart

```diff
@@ -217,4 +239,35 @@ class InspectorScreen extends Screen {
     await inspectorController?.onForceRefresh();
     refreshTreeButton.disabled = false;
   }
+
+  void _onExpandCollapseSupported() {
+    final InspectorTreeWeb detailsInspectorTree =
+        inspectorController.details.inspectorTree;
+
+    // Show the expand collapse buttons on initial load if the details tree is
+    // not empty.
+    if (detailsInspectorTree.root != null) {
+      expandCollapseButtonGroup.hidden(false);
+    }
+    // Ensure the expand collapse buttons are visible if we have a
+    // selected node.
+    inspectorController.onTreeNodeSelected.listen((_) {
+      expandCollapseButtonGroup.hidden(false);
+    });
+
+    final currentChild =
+        CoreElement.from(detailsInspectorTree.element.element.children.first);
+    detailsInspectorTree.element
+      ..clear()
+      ..add([
+        div(c: 'expand-collapse-container')
+          ..layoutHorizontal()
+          ..add([
+            div()..flex(),
+            expandCollapseButtonGroup,
+          ]),
+        // Add negative margin to offset height of expand/reset button group.
+        currentChild..element.style.marginTop = '-30px',
```

Comment:
why is this magic constant 30px? I expect what you want to do is place the expand collapse button one or two levels higher up in the tree so that you don't need this and can just absolutely position them.

Reply:
30px is the height of the button group. This as far up in the tree we can put the button group element because the next level up is the container that houses both the summary tree and the details tree. Once the screen ratio moves to below 1:1, we position the trees vertically instead of horizontally, at which point the button group would be in the wrong position.